### PR TITLE
[CELEBORN-702][DOC] Extend doc about migration from 0.2.1 to 0.3.0

### PR DESCRIPTION
**Pull request by waitinfuture**
### What changes were proposed in this pull request?
Extend doc about migration from 0.2.1 to 0.3.0. Added the following contents:

image

### Why are the changes needed?
ditto

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
No.

### CODE REVIEW

**Comment by waitinfuture** [comment]
cc @FMX

**Comment by pan3793** [comment]
Can we use [Collapsible blocks](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#collapsible-blocks) for stacktrace?

**Comment by waitinfuture** [comment]
> Can we use [Collapsible blocks](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#collapsible-blocks) for stacktrace?

done

File: docs/migration.md [hunk 1]
```diff
@@ -29,4 +29,21 @@ license: |
    users who want to use origin working dir path should set this configuration.
 
  - Since 0.3.0, configuration namespace `celeborn.ha.master` is deprecated, and will be removed in the future versions.
-   All configurations `celeborn.ha.master.*` should migrate to `celeborn.master.ha.*`.
\ No newline at end of file
+   All configurations `celeborn.ha.master.*` should migrate to `celeborn.master.ha.*`.
+
+ - When using 0.2.1 as client side and 0.3.0 as server side, you may see the following Exception in LifecycleManger's
+   log. You can safely ignore the log, it's caused by the behavior change when Master receives heartbeat from Application.
+```
```
[the remark this answers is not in the document]
Comment (waitinfuture) [reply docs/migration.md]: Seems it can only use 4 leading spaces or it will be a messy

File: docs/migration.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 hide:
-  - navigation
+- navigation
```
Comment (pan3793) [inline docs/migration.md:3]: unnecessary change

Comment (waitinfuture) [reply docs/migration.md]: fixed

File: docs/migration.md [hunk 3]
```diff
@@ -29,4 +29,21 @@ license: |
...
+    at java.io.ObjectInputStream.readEnum(ObjectInputStream.java:2157)
+    at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1662)
+    at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2430)
+    at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2354)
+    at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2212)
+    at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1668)
+    at java.io.ObjectInputStream.readObject(ObjectInputStream.java:502)
+    at java.io.ObjectInputStream.readObject(ObjectInputStream.java:460)
+    at org.apache.celeborn.common.serializer.JavaDeserializationStream.readObject(JavaSerializer.scala:76)
+    at org.apache.celeborn.common.serializer.JavaSerializerInstance.deserialize(JavaSerializer.scala:110)
```
Comment (pan3793) [inline docs/migration.md:49]:
```suggestion
    ??? Logs
        ```
        23/06/20 18:12:30 WARN TransportChannelHandler: Exception in connection from /192.168.1.16:9097
            java.io.InvalidObjectException: enum constant HEARTBEAT_FROM_APPLICATION_RESPONSE does not exist in class org.apache.celeborn.common.protocol.MessageType
            at java.io.ObjectInputStream.readEnum(ObjectInputStream.java:2157)
            at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1662)
            at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2430)
            at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2354)
            at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2212)
            at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1668)
            at java.io.ObjectInputStream.readObject(ObjectInputStream.java:502)
            at java.io.ObjectInputStream.readObject(ObjectInputStream.java:460)
            at org.apache.celeborn.common.serializer.JavaDeserializationStream.readObject(JavaSerializer.scala:76)
            at org.apache.celeborn.common.serializer.JavaSerializerInstance.deserialize(JavaSerializer.scala:110)
        ```
```

File: docs/migration.md [hunk 4]
```diff
@@ -29,4 +29,23 @@ license: |
...
 
  - Since 0.3.0, configuration namespace `celeborn.ha.master` is deprecated, and will be removed in the future versions.
-   All configurations `celeborn.ha.master.*` should migrate to `celeborn.master.ha.*`.
\ No newline at end of file
+   All configurations `celeborn.ha.master.*` should migrate to `celeborn.master.ha.*`.
+
+ - When using 0.2.1 as client side and 0.3.0 as server side, you may see the following Exception in LifecycleManger's
+   log. You can safely ignore the log, it's caused by the behavior change when Master receives heartbeat from Application.
+
+    ??? Logs
```
Comment (pan3793) [inline docs/migration.md:37]:
```suggestion
    ??? warning "logs"
```

Comment (waitinfuture) [reply docs/migration.md]: 👍 refined

File: docs/migration.md [hunk 5]
```diff
@@ -29,4 +29,23 @@ license: |
...
\ No newline at end of file
+   All configurations `celeborn.ha.master.*` should migrate to `celeborn.master.ha.*`.
+
+ - When using 0.2.1 as client side and 0.3.0 as server side, you may see the following Exception in LifecycleManger's
+   log. You can safely ignore the log, it's caused by the behavior change when Master receives heartbeat from Application.
+
+    ??? warning "logs"
+        ```
+        23/06/20 18:12:30 WARN TransportChannelHandler: Exception in connection from /192.168.1.16:9097
+            java.io.InvalidObjectException: enum constant HEARTBEAT_FROM_APPLICATION_RESPONSE does not exist in class org.apache.celeborn.common.protocol.MessageType
```
Comment (pan3793) [inline docs/migration.md:40]:
```suggestion
        23/06/20 18:12:30 WARN TransportChannelHandler: Exception in connection from /192.168.1.16:9097
        java.io.InvalidObjectException: enum constant HEARTBEAT_FROM_APPLICATION_RESPONSE does not exist in class org.apache.celeborn.common.protocol.MessageType
```

**Review by pan3793 — APPROVED** [review]
